rasa/nlu/test.py

```diff
@@ -1246,54 +1245,44 @@ def get_eval_data(
 
     intent_results, entity_results, response_selection_results = [], [], []
 
-    response_labels = [
+    response_labels = {
         e.get(INTENT_RESPONSE_KEY)
         for e in test_data.intent_examples
         if e.get(INTENT_RESPONSE_KEY) is not None
-    ]
+    }
     intent_labels = [e.get(INTENT) for e in test_data.intent_examples]
```

Comment:
optional: Should this also be a `set` then?
```suggestion
    intent_labels = {e.get(INTENT) for e in test_data.intent_examples}
```